docs/src/apimanual.md

```diff
@@ -469,59 +484,115 @@ non-global tree search solvers like
+want to solve a binary-constrained knapsack problem:
+`max c'x: w'x <= C, x binary`. Suppose we choose GLPK:
 ```julia
-using MathOptInterface
-const MOI = MathOptInterface
 using GLPK
-
-# Solves the binary-constrained knapsack problem:
-# max c'x: w'x <= C, x binary using GLPK.
-
+optimizer = GLPK.Optimizer()
+```
+We first define the constants of the problem:
+```jldoctest knapsack; setup = :(optimizer = MOI.Utilities.MockOptimizer(MOI.Utilities.Model{Float64}()); MOI.Utilities.set_mock_optimize!(optimizer, mock -> MOI.Utilities.mock_optimize!(mock, ones(3))))
 c = [1.0, 2.0, 3.0]
 w = [0.3, 0.5, 1.0]
 C = 3.2
 
-num_variables = length(c)
-
-optimizer = GLPK.Optimizer()
+num_variables_to_create = length(c)
 
-# Create the variables in the problem.
-x = MOI.add_variables(optimizer, num_variables)
+# output
 
-# Set the objective function.
+3
+```
+We create the variables of the problem and set the objective function:
+```jldoctest knapsack
+x = MOI.add_variables(optimizer, num_variables_to_create)
 objective_function = MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(c, x), 0.0)
 MOI.set(optimizer, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
         objective_function)
 MOI.set(optimizer, MOI.ObjectiveSense(), MOI.MAX_SENSE)
 
-# Add the knapsack constraint.
+# output
+
+MAX_SENSE::OptimizationSense = 1
+```
+We add the knapsack constraint and integrality constraints:
+```jldoctest knapsack
 knapsack_function = MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(w, x), 0.0)
 MOI.add_constraint(optimizer, knapsack_function, MOI.LessThan(C))
-
-# Add integrality constraints.
-for i in 1:num_variables
+for i in 1:num_variables_to_create
     MOI.add_constraint(optimizer, MOI.SingleVariable(x[i]), MOI.ZeroOne())
 end
 
-# All set!
+# output
+
+```
+We are all set! We can now call [`optimize!`](@ref) and wait for the solver to
+find the solution:
+```jldoctest knapsack
 MOI.optimize!(optimizer)
 
-termination_status = MOI.get(optimizer, MOI.TerminationStatus())
-obj_value = MOI.get(optimizer, MOI.ObjectiveValue())
-if termination_status != MOI.OPTIMAL
-    error("Solver terminated with status $termination_status")
-end
+# output
+
+```
+The first thing to check after optimization is why the solver stopped, e.g.,
+did it stop because of a time limit or did it stop because it found the optimal
+solution ?
+```jldoctest knapsack
+MOI.get(optimizer, MOI.TerminationStatus())
+
+# output
+
+
+OPTIMAL::TerminationStatusCode = 1
+```
+It found the optimal solution! Now let's see what is that solution.
+But first, let's check if it has more than one solution to share:
+```jldoctest knapsack
+MOI.get(optimizer, MOI.ResultCount())
+
+# output
+
+1
+```
+Only one.
 
-@assert MOI.get(optimizer, MOI.ResultCount()) > 0
+!!! note
+    While the value of `MOI.get(optimizer, MOI.ResultCount())` is often one, it
+    is important to check its value in order to write a robust code. For
+    instance, when the problem is unbounded, the solver might return two
```

Comment:
Which solvers, if any, actually do this?

Reply:
[VRepOptimizer](https://github.com/JuliaPolyhedra/Polyhedra.jl/blob/master/src/vrep_optimizer.jl) does not currently do it but it should. Apart from that, not sure. I chose the example that makes most sense in theory. Note that the example that `ResultCount` could be zero if it is infeasible is already discussed in `apimanual`.

Reply:
The theory doesn't really matter if there are zero solvers that do this. Another example in theory is MIP solvers that could return multiple suboptimal solutions, but AFAIK this isn't implemented anywhere yet. Maybe we could remove the discussion about result counts from the intro example.